fe/fe-core/src/main/java/org/apache/doris/qe/SimpleScheduler.java

```diff
@@ -134,8 +134,8 @@ public static TNetworkAddress getHost(ImmutableMap<Long, Backend> backends,
         Map.Entry<Long, Backend> backendEntry = backends.entrySet().stream().skip(id).filter(
                 e -> isAvailable(e.getValue())).findFirst().orElse(null);
         if (backendEntry == null && id > 0) {
-            backendEntry = backends.entrySet().stream().filter(
-                e -> isAvailable(e.getValue())).limit(id).findFirst().orElse(null);
+            backendEntry = backends.entrySet().stream().limit(id).filter(
```

Comment:
Why first limit then filter?

Reply:
just keep same with the origin algorithm.